Demo.Web/App_Start/AutofacConfig.cs

```diff
@@ -27,22 +27,30 @@ public static void Register(HttpConfiguration config)
                 .Classes()
                 .ForAll(t =>
                 {
-                    if (t.IsGenericType)
-                        builder
-                            .RegisterGeneric(t)
-                            .AsImplementedInterfaces();
-                    else
-                    {
-                        builder
-                            .RegisterType(t)
-                            .AsSelf();
-                        t
-                            .GetInterfaces()
-                            .ForAll(i => builder
-                                .RegisterType(WrapperFactory.Emit(i))
-                                .WithParameter((pi, ctx) => true, (pi, ctx) => ctx.Resolve(t))
-                                .AsImplementedInterfaces());
-                    }
+                    builder
+                        .RegisterType(t)
+                        .AsImplementedInterfaces()
+                        .InstancePerRequest();
+                });
```

Comment:
@johnnyoshika I would like to keep this Wrapper generation logic - there is a very good probability we might need it. Let's commit this simplified version, we will restore the original one if event.Implement() functionality is required.